src/stan/model/indexing/assign_varmat.hpp

```diff
@@ -353,7 +490,7 @@ inline void assign(Mat1&& x, const Mat2& y, const char* name,
       for (int i = assign_rows - 1; i >= 0; --i) {
         if (likely(x_row_idx[i] != -1)) {
           prev_vals.coeffRef(i, j) = x.vi_->val_(x_row_idx[i], x_col_idx[j]);
-          y_vals(i, j) = y.vi_->val_(i, j);
+          y_vals(i, j) = y_val(i, j);
```

Comment:
Unrelated to changes in this PR, but would not it be better to use `.coeff()` and `.coeffRef()` here?

Reply:
Yeah I went through and fixed these up!